rmw_fastrtps_shared_cpp/src/demangle.cpp

```diff
@@ -32,19 +34,21 @@ _demangle_if_ros_topic(const std::string & topic_name)
 std::string
 _demangle_if_ros_type(const std::string & dds_type_string)
 {
-  std::string substring = "::msg::dds_::";
+  std::string substring = "dds_::";
   size_t substring_position = dds_type_string.find(substring);
   if (
-    dds_type_string[dds_type_string.size() - 1] == '_' &&
-    substring_position != std::string::npos)
+    dds_type_string[dds_type_string.size() - 1] != '_' ||
```

Comment:
Small performance nit: you can skip the `find` completely if the last character isn't `_`.

Reply:
See 3f34667